src/main/java/org/assertj/core/util/NaturalOrderComparator.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2021 the original author or authors.
+ */
+package org.assertj.core.util;
+
+import java.nio.file.Path;
+import java.util.Comparator;
+
+import static java.lang.String.format;
+
+public class NaturalOrderComparator<T extends Comparable<? super T>> extends NullSafeComparator<T> {
+
+  private String description;
+
+  public NaturalOrderComparator(Class<T> clazz) {
+    this.description = format("lexicographic comparator (%s naturalComparator)", clazz.getSimpleName());
```

Comment:
What about the following?
```suggestion
    this.description = format("lexicographic comparator (%s natural order)", clazz.getSimpleName());
```

Reply:
happy with that, @VanRoy I have started integrating your PR so I'll make the change (you'll get full credit of course ;-) )

Reply:
one thing that I'm changing is removing `lexicographic ` from here since it's for Path only and subclassing `NaturalOrderComparator` with a `NaturalPathComparator`

Reply:
Indeed, it's better like that. 
Thanks.